src/wallet/rpcwallet.cpp

```diff
@@ -1053,27 +1071,39 @@ static UniValue ListReceived(const CWallet& wallet, const UniValue& params, bool
 
     bool has_filtered_address = false;
     CTxDestination filtered_address = CNoDestination();
-    if (!by_label && params.size() > 3) {
+    if (!by_label && !params[3].isNull() && !params[3].get_str().empty()) {
         if (!IsValidDestinationString(params[3].get_str())) {
             throw JSONRPCError(RPC_WALLET_ERROR, "address_filter parameter was invalid");
         }
         filtered_address = DecodeDestination(params[3].get_str());
         has_filtered_address = true;
     }
 
+    // getreceivedbyaddress has address filter at 3rd index, so next params index is 4
+    // getreceivedbylabel has no filter parameter, so next params index is 3
+    int next_index{by_label ? 3 : 4};
+    bool include_immature{!params[next_index].isNull() && params[next_index].get_bool()};
```

Comment:
C++ is a strongly typed language - we should use types in the function signatures to catch logic errors at compilation rather than pass through opaque structures.

Can you update the function signature of `ListReceived()` to explicitly take the argument types?